examples/pipelines/datacomp/components/detect_text/src/main.py

```diff
@@ -0,0 +1,194 @@
+"""This component detects text in images using an mmocr model.
+
+Based on notebook: Run inference with FAST for text detection.ipynb.
+"""
+import io
+import logging
+import typing as t
+
+import numpy as np
+import pandas as pd
+import torch
+from fondant.component import PandasTransformComponent
+from PIL import Image
+
+logger = logging.getLogger(__name__)
+
+from mmengine.config import Config
+
+# import subprocess
+# # make sure to compile the postprocessing scripts
+# subprocess.call(["sh", "./compile.sh"])  # nosec
+
+from models import build_model
+from models.utils import fuse_module, rep_model_convert
+
+from augmentations import SquarePadResizeNorm
+
+from huggingface_hub import hf_hub_download
+
+
+def poly2bbox(polygon: np.array) -> np.array:
+    """Converting a polygon to a bounding box.
+
+    Args:
+         polygon (ArrayLike): A polygon. In any form can be converted
+             to an 1-D numpy array. E.g. list[float], np.ndarray,
+             or torch.Tensor. Polygon is written in
+             [x1, y1, x2, y2, ...].
+
+     Returns:
+         np.array: The converted bounding box [x1, y1, x2, y2]
+    """
+    if len(polygon) % 2 != 0:
+        raise ValueError("Not ok")
+    polygon = np.array(polygon, dtype=np.float32)
+    x = polygon[::2]
+    y = polygon[1::2]
+    return np.array([min(x), min(y), max(x), max(y)])
+
+
+def process_image(image: bytes, *, image_transform, device: str) -> torch.Tensor:
+    """
+    Process the image to a tensor.
+
+    Args:
+        image: The input image as a byte string.
+        image_transform: The image transformation to apply.
+        device: The device to move the transformed image to.
+    """
+
+    def load(img: bytes) -> Image:
+        """Load the bytestring as an image."""
+        bytes_ = io.BytesIO(img)
+        return Image.open(bytes_).convert("RGB")
+
+    def transform(img: Image) -> torch.Tensor:
+        """Transform the image to a tensor and move it to the specified device."""
+        return image_transform(img)[0][0].to(device)
+
+    return transform(load(image))
+
+
+def detect_text_batch(
+    image_batch: pd.DataFrame,
+    *,
+    cfg,
+    model,
+) -> pd.Series:
+    """Detext text on a batch of images."""
+    imgs = torch.stack(list(image_batch), dim=0)
+    batch_size = imgs.shape[0]
+    # TODO fix this, make this component also take width and height
+    # as input in spec to process arbitrarly sized images
+    img_metas = {
+        "filename": [None for i in range(batch_size)],
+        "org_img_size": torch.ones((batch_size, 2)).long() * 256,
+        "img_size": torch.ones((batch_size, 2)).long() * 512,
+    }
+
+    data = dict()
+    data["imgs"] = imgs
+    data["img_metas"] = img_metas
+    data.update(dict(cfg=cfg))
+
+    # forward
+    with torch.no_grad():
+        outputs = model(**data)
+
+    # get cropped images
+    boxes_batch = []
```

Comment:
Can you split `detect_text_image` into two functions similar to  `process_image`, one for detection and other for cropping? Helps in readability